dhis-2/dhis-api/src/main/java/org/hisp/dhis/message/MessageSender.java

```diff
@@ -71,4 +74,14 @@ ListenableFuture<OutboundMessageResponseSummary> sendMessageBatchAsync(
 
   /** To check if given service is configured and ready to use. */
   boolean isConfigured();
+
+  default List<OutboundMessage> getMessagesByEmail(String recipient) {
+    return List.of();
+  }
+
+  default void clearMessages() {}
+
+  default List<OutboundMessage> getAllMessages() {
+    return List.of();
+  }
```

Comment:
wondering what these new default methods are for? are they used anywhere?
should a MessageSender just send messages, or get messages as well? :)

Reply:
i've seen it used in the FakeMessageSender while looking through the rest of the PR. This seems like a code smell if it's only used for tests.

Reply:
This is currently a workaround for fixing the issue of overriding a bean in test environment. As `@Qualifer` doesn't work for that purpose.  I can give an explanation in next week meeting.
Basically in test we have `FakeMessageSender` `SmsMessageSender` and `EmailMessageSender`.
We want the `FakeMessageSender` to override both `SmsMessageSender` and `EmailMessageSender`.  
But `FakeMessageSender` has specific methods `getAllMessages()`, which then can't be casted back to the others messageSenders.

Reply:
In Spring 5 you can override a bean by declaring same bean name. 
But in spring 6, it doesn't work anymore. Need to implement same interface.